### PR TITLE
Add CircleCI integration

### PR DESCRIPTION
**Pull request by planteater**
1. Reorganized and renamed the “How to use CodeStream” section, as the current paradigm didn’t work anymore. Originally the titles were oriented around walking the user through a workflow, but as we add more features into CS that model is breaking down.
2. Added new CI/CD article to that section.
3. Added expander for Blame Hovers to the UI Overview article. Also, added CI/CD in bullet list in Sidebar section.

### CODE REVIEW

File: src/nav/codestream.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ pages:
         path: /docs/codestream/troubleshooting/git-issues
       - title: GitHub organizations
         path: /docs/codestream/troubleshooting/github-org-repos
-      - title: GitLab Self-Managed version
+      - title: GitLab self-managed version
```
Comment (planteater) [inline src/nav/codestream.yml:50]: @zuluecho9 - "GitLab Self-Managed" is a product name and needs to be capitalized.

Comment (zuluecho9) [reply src/nav/codestream.yml:50]: i saw it lower case here and elsewhere: https://docs.gitlab.com/ee/subscriptions/self_managed/. are you seeing it capitalized elsewhere? @planteater

Comment (planteater) [reply src/nav/codestream.yml:50]: Looks like GL is inconsistent in their display, although that feels like a recent thing. I think we should stick with upper either way though, as we use that all throughout the product. And it's simply clearer that it's a product name.

Comment (zuluecho9) [reply src/nav/codestream.yml:50]: ok

**Review by zuluecho9 — APPROVED** [review]

**Comment by zuluecho9** [comment]
sorry @planteater restarting the build, think it timed out. hopefully will go out in morning.

**Comment by planteater** [comment]
@zuluecho9 - Seems like things are still stuck here?

**Comment by planteater** [comment]
There some urgency in having this live tomorrow if at all possible. Thanks!

**Comment by zuluecho9** [comment]
sorry @planteater, thought I merged that. merging now

**Pull request merged**